homeassistant/components/green_planet_energy/coordinator.py

```diff
@@ -0,0 +1,46 @@
+"""Data update coordinator for Green Planet Energy."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from greenplanet_energy_api import (
+    GreenPlanetEnergyAPI,
+    GreenPlanetEnergyAPIError,
+    GreenPlanetEnergyConnectionError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GreenPlanetEnergyUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Class to manage fetching data from Green Planet Energy API."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
```

Comment:
```suggestion
    def __init__(self, hass: HomeAssistant, config_entry: GreenPlanetEnergyConfigEntry) -> None:
```
You might need to define the `type` in the coordinator file